src/lib/warn-upload-overwrite.js

```diff
@@ -0,0 +1,73 @@
+const fs = require('./sync-fs');
+const jsonDiff = require('json-diff');
+const readline = require('readline-sync');
+
+const question = 'You are trying to modify a configuration that has been modified since your last upload. Do you want to?';
+const responseChoicesWithoutDiff = [
+  'Overwrite the changes', 
+  'Abort so that you can update the configuration'
+];
+const responseChoicesWithDiff = responseChoicesWithoutDiff.concat([ 'View diff' ]);
+
+const preUpload = async (projectDir, db, localDoc) => {
+
+  // Pull remote _rev
+  let remoteRev;
+  let remoteDoc;
+  try {
+    remoteDoc = await db.get(localDoc._id);
+    remoteRev = remoteDoc._rev;
+  } catch (e) {
+    // continue regardless of error
+  }
+
+  // Pull local _rev
+  let localRev;
+  try {
+    localRev = fs.read(`${projectDir}/._revs/${localDoc._id}`);
+  } catch (e) {
+    // continue regardless of error
+  }
+
+  // Compare _revs
+  // If _revs are different, show prompt
+  if (!localRev) {
+    if (!readline.keyInYN(`We can't determine if you're going to overwrite someone else's changes or not. Would you like to proceed?`)) {
+      throw new Error('configuration modified');
+    }
+  } else if (localRev !== remoteRev) {
+    let diff; 
+    if (localDoc.settings && remoteDoc.settings) {
```

Comment:
This is great. I think it just works for the settings doc right? Can you think of a way to get a diff to work for the other types of docs too? For example, actually comparing the attachments etc?

Also I think we should do the diff check before showing the warning when there's no local rev. Ideally the first time the configuration is synced it will detect which docs are the same, create the localRev, and move on.

Reply:
hey @garethbowen maybe I don't understand your comment but isn't what https://github.com/medic/medic-conf/pull/196/files/51a2f9f2f6854b8392ff894040f7b8efc1158727#diff-a10d0dcee98f0f19513039f23c2fc71cR43 is doing .. it does the diff for other documents

Reply:
Ahh, right. Sorry I don't know how i missed that!

In that case I think it'd be cleaner to check the _id, because the settings doc always (well, for a long time anyway) has the _id `settings`. That way it won't accidentally fire for other docs. Does that make sense?

Reply:
yes, that's right